### DISCUSSION TITLE
permanent variables p:vname issue/question

### DISCUSSION THREAD

**Discussion by puskyer** [arendst/Tasmota#16291] [Q&A] [answered]
So I was using some **p:vname** in my script, when trying to use them in  "print" commands or "if than" conditions they do not work.

example:

`>D
p:maxh=75
debug=75

>T
print %debug%  %maxh%
if debug==maxh
then
print used ram size %ram% 
print cpu frequency %freq% 
print stack size %stack% 
print free heap size %heap%
endif`

the output is the following and the if statement never gets executed.
09:12:13.486 75.00  0.00
09:12:14.688 75.00  0.00

the second value is the **p:vname** and should be 75?

if I change the **maxh** to just a variable, then my print and condition work??

`>D
maxh=75
debug=75

>T
print %debug%  %maxh%
if debug==maxh
then
print used ram size %ram% 
print cpu frequency %freq% 
print stack size %stack% 
print free heap size %heap%
endif`

09:13:03.968 75.00  75.00
09:13:03.971 used ram size 2657.00 
09:13:03.973 cpu frequency 80.00 
09:13:03.975 stack size 3024.00 
09:13:03.977 free heap size 23360.00

What am I doing wrong or are p:vname only meant to set other variables and not be used in the script?

thank you

**Comment by gemu2015** [comment, marked as answer]
it makes no sense to preset permanent variables in >D. they are set from flash before starting the script  with the values stored during runtime and are saved on restart or on cmd 'svars'
they just remember values between restarts and power off.
example:
\>D
p:maxh=0

\>S
if chg[maxh]>0 {
svars
}

\>W
sl(0 100 maxh "0" "value %0maxh%" "100")

allows to set maxh between 0 and 100 with a slider and remembers last value

**Reply by puskyer** [reply]
Hi @gemu2015 

Thank you for the reply, 

That makes more sense to me now, I was thinking they were like "constant"  values that I did not want to change.

cheers,
Pasquale
